lib/classes/Service.js

```diff
@@ -149,16 +149,27 @@ class Service {
         // setup function.name property
         const stageNameForFunction = options.stage || this.provider.stage;
         _.forEach(that.functions, (functionObj, functionName) => {
+          const targetFunction = that.functions[functionName];
           if (!functionObj.events) {
-            that.functions[functionName].events = [];
+            targetFunction.events = [];
           }
           if (!_.isArray(functionObj.events)) {
             throw new SError(`Events for "${functionName}" must be an array,` +
                              ` not an ${typeof functionObj.events}`);
           }
 
+          // Translate HTTP events written in string notation to object notation
+          _.forEach(functionObj.events, (event, index) => {
```

Comment:
questions:

a/ what is the AWS API Gateway plugin? do you refer to lib/plugins/aws/deploy/compile/events/apiGateway?

b/ I wanted to make this normalization as early as possible in the program's lifecycle so that all plugins will see the fixed ("normalized") form. I may be missing something but I am worried that if I put this in a plugin then there is a change that other plugins (which may need to insepct the HTTP events) will see the raw ("non-normalized") form. As a quick data point, I found [aws/info/index.js](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/info/index.js#L179-L185)(note to self: I'll need to change it too as part of this PR....)

this boils down to this question: Is there a way to do that early in the lifecylce and yet have the code in a plugin? perhaps there is a "service-filed-loaded-hook" which plugins can utilize to carry out such normalization tasks?